discoverable-client/src/main/java/org/zowe/apiml/client/api/graphql/GraphQLExceptionResolver.java

```diff
@@ -0,0 +1,27 @@
+package org.zowe.apiml.client.api.graphql;
+
+import graphql.GraphQLError;
+import graphql.GraphqlErrorBuilder;
+import graphql.schema.DataFetchingEnvironment;
+import org.springframework.graphql.execution.DataFetcherExceptionResolverAdapter;
+import org.springframework.graphql.execution.ErrorType;
+import org.springframework.stereotype.Component;
+import org.zowe.apiml.client.exception.BookAlreadyExistsException;
+
+@Component
+public class GraphQLExceptionResolver extends DataFetcherExceptionResolverAdapter {
+
+    @Override
+    protected GraphQLError resolveToSingleError(Throwable ex, DataFetchingEnvironment env) {
+        if (ex instanceof BookAlreadyExistsException) {
+            return GraphqlErrorBuilder.newError()
+                .errorType(ErrorType.BAD_REQUEST)
+                .message(ex.getMessage())
+                .path(env.getExecutionStepInfo().getPath())
+                .location(env.getField().getSourceLocation())
+                .build();
+        } else {
+            return null;
```

Comment:
Could we maybe rather return another response error, i.e. Internal Error, in this case?